### PR TITLE
[gl] Add Instance struct in web version to align API with other backends.

### PR DESCRIPTION
**Pull request by VincentFTS**
PR checklist:
- [X] `make` succeeds (on macOS)
- [X] `make reftests` succeeds
- [X] tested examples with the following backends: quad example with gl backend with wasm target arch (no windows nor linux available to test gl backend there)

### CODE REVIEW

**Comment by VincentFTS** [comment]
I did not protect `create_surface()` from creating one more surface when one already exists since no Error seems to be relevant.

File: src/backend/gl/src/window/web.rs [hunk 1]
```diff
@@ -15,38 +15,68 @@ pub struct Swapchain {
...
+                swapchain: None,
+                renderbuffer: None,
+            })
         } else {
             unreachable!()
         }
     }
 
+    unsafe fn destroy_surface(&self, _surface: Surface) {
+        // TODO: Implement Surface cleanup
```
Comment (kvark) [inline src/backend/gl/src/window/web.rs:68]: can just make the canvas to `None`?

**Review by kvark — COMMENTED** [review]
This looks great! It's the last piece of the puzzle to get the API aligned between all the backends. Just a few notes before we go.

**Review by kvark — APPROVED** [review]
Well done, thank you!
bors r+

**Comment by VincentFTS** [comment]
My pleasure !

**Pull request merged**